code/game/objects/effects/temporary_visuals/miscellaneous.dm

```diff
@@ -228,4 +228,65 @@
 	randomdir = FALSE
 	pixel_y = -16
 	pixel_x = -16
-	duration = 20
\ No newline at end of file
+	duration = 20
+
+/obj/effect/temp_visual/bleed
+	name = "bleed"
+	icon = 'icons/effects/bleed.dmi'
+	icon_state = "bleed0"
+	duration = 10
+	var/shrink = TRUE
+
+/obj/effect/temp_visual/bleed/Initialize(mapload, atom/size_calc_target)
+	. = ..()
+	var/size_matrix = matrix()
+	if(size_calc_target)
+		layer = size_calc_target.layer + 0.01
+		var/icon/I = icon(size_calc_target.icon, size_calc_target.icon_state, size_calc_target.dir)
+		size_matrix = matrix() * (I.Height()/world.icon_size)
+		transform = size_matrix //scale the bleed overlay's size based on the target's icon size
+	var/matrix/M = transform
+	if(shrink)
+		M = size_matrix*0.1
+	else
+		M = size_matrix*2
+	animate(src, alpha = 20, transform = M, time = duration, flags = ANIMATION_PARALLEL)
```

Comment:
When you start dealing with crazy math like this I'm always like "well ok then, I guess I"ll have to test it to see if it looks ok".

Goon's animation library is just as confusing.

Reply:
The matrix ain't no joke